internal/async/wait.go

```diff
@@ -0,0 +1,75 @@
+package async
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	defaultInterval = time.Second
+	defaultTimeout  = time.Minute * 5
+)
+
+// WaitOptions defines the waiting options.
+type WaitSyncConfig struct {
```

Comment:
WaitSyncRequest?

Reply:
`WaitSync` very abstracted and could be used to other use cases than waiting for a request. Therefore, I think that the concept of "config" is better than "request".